gnd/src/main/java/com/google/android/gnd/repository/OfflineAreaRepository.java

```diff
@@ -186,4 +211,27 @@ public Completable addAreaAndEnqueue(LatLngBounds bounds) {
                     .filter(tileSource -> tileSource.getState() == TileSource.State.DOWNLOADED)
                     .collect(toImmutableSet()));
   }
+
+  /** Delete an offline area and the unique tile sources associated with it. */
+  public Completable deleteArea(String offlineAreaId) {
+    return localDataStore
+        .getOfflineAreaById(offlineAreaId)
+        .flatMapMaybe(this::getIntersectingDownloadedTileSourcesOnce)
+        .flatMapCompletable(
```

Comment:
Same here.. Any way to make this more readable?

Reply:
b86a87a